### ISSUE TITLE
rootpath produces leading ./ which causes rlocation to fail

### ISSUE THREAD

**Issue by aherrmann** [bazelbuild/bazel#10670]
### Description of the problem:

Applying `$(execroot)` (or `$(location)`, or `$(rootpath)`) on a top-level source file, produces a path with a leading `./`. This leading `./` in turn causes bash `$(rlocation)` to fail to find the sourcefile.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Create the following project
```
$ touch WORKSPACE resource
$ cat >test.sh <<EOF
# Copy-pasted from the Bazel Bash runfiles library v2.
set -uo pipefail; f=bazel_tools/tools/bash/runfiles/runfiles.bash
source "${RUNFILES_DIR:-/dev/null}/$f" 2>/dev/null || \
  source "$(grep -sm1 "^$f " "${RUNFILES_MANIFEST_FILE:-/dev/null}" | cut -f2- -d' ')" 2>/dev/null || \
  source "$0.runfiles/$f" 2>/dev/null || \
  source "$(grep -sm1 "^$f " "$0.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
  source "$(grep -sm1 "^$f " "$0.exe.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
  { echo>&2 "ERROR: cannot find $f"; exit 1; }; f=; set -e
# --- end runfiles.bash initialization v2 ---

set -ex
# This fails due to leading `./` on `$1`.
RESOURCES=$(rlocation $TEST_WORKSPACE/$1)
# This works around the issue by removing the leading `./`.
# RESOURCES=$(rlocation $TEST_WORKSPACE/${1#./})
echo $RESOURCES
EOF
$ chmod +x test.sh
$ cat >BUILD.bazel <<EOF
sh_test(
    name = "test",
    srcs = ["test.sh"],
    args = ["$(rootpath //:resource)"],
    data = ["resource"],
    deps = ["@bazel_tools//tools/bash/runfiles"],
)
EOF
```
Execute the test as follows and observe the error:
```
$ bazel test //:test --test_output=streamed
WARNING: Streamed test output requested. All tests will be run locally, without sharding, one at a time
INFO: Analyzed target //:test (0 packages loaded, 0 targets configured).
INFO: Found 1 test target...
+++ rlocation __main__/./resource
+++ [[ '' == 1 ]]
+++ [[ __main__/./resource =~ ^/[^/].* ]]
+++ [[ __main__/./resource == ../* ]]
+++ [[ __main__/./resource == */.. ]]
+++ [[ __main__/./resource == ./* ]]
+++ [[ __main__/./resource == */./* ]]
+++ [[ '' == 1 ]]
+++ return 1
++ RESOURCES=
FAIL: //:test (see /home/aj/.cache/bazel/_bazel_aj/c7e97ee9cc7bd4601835d9d415c1e2e3/execroot/__main__/bazel-out/k8-fastbuild/testlogs/test/test.log)
Target //:test up-to-date:
  bazel-bin/test
INFO: Elapsed time: 0.457s, Critical Path: 0.24s
INFO: 2 processes: 2 linux-sandbox.
INFO: Build completed, 1 test FAILED, 2 total actions
//:test                                                                  FAILED in 0.1s
  /home/aj/.cache/bazel/_bazel_aj/c7e97ee9cc7bd4601835d9d415c1e2e3/execroot/__main__/bazel-out/k8-fastbuild/testlogs/test/test.log

INFO: Build completed, 1 test FAILED, 2 total actions
```

### What operating system are you running Bazel on?

Linux

### What's the output of `bazel info release`?

`release 2.0.0`

### Any other information, logs, or outputs that you want to share?

As indicated in `test.sh` above, the issue can be worked around by stripping the leading `./`.

**Comment by dmivankov** [comment]
still present in 4.1.0

**Comment by dmivankov** [comment]
also slightly related to https://github.com/bazelbuild/bazel/issues/12198

**Comment by fmeum** [comment]
This is fixed, use `$(rlocationpath ...)` instead (available as of 5.4.0).
